service/raft/group0_state_machine.cc

```diff
@@ -377,6 +377,49 @@ future<> group0_state_machine::merge_and_apply(group0_state_machine_merger& merg
     co_await _sp.mutate_locally({std::move(history)}, nullptr);
 }
 
+static void ensure_group0_schema(const group0_command& cmd, const replica::database& db) {
+    auto validate_schema = [&db](const std::vector<canonical_mutation>& mutations) {
+        for (const auto& mut : mutations) {
+            // Get the schema for the column family
+            auto schema = db.find_schema(mut.column_family_id());
+            if (!schema) {
+                on_internal_error(slogger, "ensure_group0_schema: schema not found");
+            }
+
+            static const db::config cfg;
+            static const auto keyspace_tables = db::system_keyspace::all_tables(cfg);
+
+            // only check the "standard" known keyspace tables
+            const bool is_table_keyspace = std::ranges::any_of(keyspace_tables, [schema](const auto& table) {
+                return table->ks_name() == schema->ks_name() && table->cf_name() == schema->cf_name();
+            });
+
+            if (is_table_keyspace && !schema->static_props().is_group0_table) {
```

Comment:
what about system_schema tables?

in fact I don't understand why restrict the check to only tables in the `system` keyspace.

Reply:
it was needed to not fail if someone adds a table and uses a group0 mutations e.g. in a test (like for example done in the boost/group0_test that started failing after adding this check)

but it is actually possible to add the group0 flag in the test, so that solution is in fact better (plus simplifies this check)